docs/web-testing-library/intro.mdx

```diff
@@ -0,0 +1,49 @@
+---
+id: intro
+title: Introduction
+---
+
+## The problem
+
+You want to write maintainable tests for the
+[Web Platform](https://proposal-common-min-api.deno.dev/) or React Native. As a
+part of this goal, you want your tests to avoid including implementation details
+of your components and rather focus on making your tests give you the confidence
+for which they are intended. As part of this, you want your testbase to be
+maintainable in the long run so refactors of your components (changes to
+implementation but not functionality) don't break your tests and slow you and
+your team down.
+
+## This solution
+
+The `Web Testing Library` is a very light-weight solution for testing code that
+runs on the Web Platform (a browser, [Node.js][node], [Deno](https://deno.land/)
+etc) or React Native. The main utilities it provides involve querying the DOM
+for nodes in a way that's similar to how the user finds elements on the page. In
+this way, the library helps ensure your tests give you confidence in your UI
+code. The `DOM Testing Library`'s primary guiding principle is:
+
+> The more your tests resemble the way your software is used, the more
+> confidence they can give you.
+
+We try to only expose methods and utilities that encourage you to write tests
+that closely resemble how your code is used on the Web Platform or React Native.
+
+Utilities are included in this project based on the following guiding
+principles:
+
+1. They should be usable on both the Web Platform **and** React Native
```

Comment:
The name "web testing library" is a bit confusing if it also works for React Native, which is definitely not the "web". Is the goal of this library to provide a library- and platform- agnostic API to replace some parts of the existing ecosystem?

Reply:
The goal is not to be platform agnostic (see e.g. ´setTimeout´) but offer shared utilities that are used on the most common platforms

Reply:
I see. In that case I'd suggest naming it something that is less similar to the framework wrappers that implement a full element traversal API, and more similar to the util libraries like user-event.

Reply:
React Native is now official part of the web-interoperable runtimes: https://runtime-keys.proposal.wintercg.org/#react-native

Going to adjust the wording to use "web interoperable" instad of just "web" to make it less similar.

Reply:
Any reason not to just put `waitFor` in its own package?